### PR TITLE
Pill module

### LINKED ISSUE #671: Button: separate pill styles into their own module
As part of DS6.5, the pill is going to be refined for a more specific set of use cases for filtering only. The concepts of primary and secondary will no longer apply, and there will be other fundamental differences, therefore I think it's a good time to do this separation.

For this task let's just separate the pill styles into their own module with no refactoring, mixins or API changes. Yes, this means there will be a little bit of duplication of styles in the button module and pill module, but this change will just set us up nicely for v9.0.0.

### PR DESCRIPTION
**Pull request by seangates**
## Description
- creates a new pill module
- adds button mixins
- adds button variables
- whole new section for pill in the documentation

## Context
Further work to split out modules and move to a more modular/mixin approach.

## References
Closes #671 

## Screenshots

image

image

### CODE REVIEW

**Comment by ianmcburnie** [comment]
@agliga @seangates I piggy-backed onto this PR and increased its scope slightly by adding aliases for selected and unselected. I believe that's what they should always have been called, rather than primary and secondary.

Screen Shot 2019-07-24 at 10 02 16 AM

File: browser.json [hunk 1]
```diff
@@ -17,11 +17,13 @@
     "./label",
     "./link",
     "./listbox",
+    "./listbox-button",
```
Comment (ianmcburnie) [inline browser.json]: Nice catch!

File: src/less/pill/ds4/pill.less [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+button.pill--regular,
+a.fake-pill--regular {
+    &:hover,
+    &:focus {
+        border-color: @ds4-color-core-gray-spanish;
+    }
+}
+
+button.pill--primary,
```
Comment (agliga) [inline src/less/pill/ds4/pill.less]: Should we nest the properties here? I see in some places you did and some places didnt. Or is it better to be verbose?
Like: 
button.pill,
a.fake-pill {
   &--primary,
   &--secondary {
   }
}

File: src/less/pill/ds6/pill.less [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    &:focus,
+    &:hover,
+    &:active {
+        background-color: @ds6-color-b7;
+        border-color: @ds6-color-b7;
+    }
+}
+
+button.pill--secondary[disabled],
```
Comment (agliga) [inline src/less/pill/ds6/pill.less]: Again, maybe nest the selectors so it's not so verbose?

**Comment by ianmcburnie** [comment]
@agliga As a heads up, I asked Sean to make minimal changes to make the diff easier. So he probably just left those like they were on purpose. The plan is for us to sweep through all of these modules we've split up again, one more time, to fix up consistency issues that you mention.

**Comment by ianmcburnie** [comment]
Drat! Only just spotted that some files were mysteriously removed in this PR:

Screen Shot 2019-07-24 at 11 23 08 AM

I'll replace them now...

**Pull request merged**